### PR TITLE
Send search click analyitcs

### PR DESCRIPTION
**Pull request by shlomi-d-walkme**

### CODE REVIEW

File: src/components/SearchResults/SearchResultItem.js [hunk 1]
```diff
@@ -4,10 +4,9 @@ import walkme from '@walkme/sdk';
 import classes from './styles.module.scss';
 import SearchResultIcon from './SearchResultIcon';
 
-export default function SearchResultItem({ node }) {
+export default function SearchResultItem({ node, wmSearch, index: key }) {
```
Comment (DavidHandler) [inline src/components/SearchResults/SearchResultItem.js:7]: Does  `SearchResultItem` should be used to show only `wmSearch` results, while other search apps will implement other components for UI? Or it should be a unified component for all apps with a well defined API?

File: src/components/SearchResults/helpers.js [hunk 2]
```diff
@@ -34,11 +36,41 @@ export function findInUiTree(searchTerm, uiTree) {
...
   try {
-    const searchResults = await wmSearch.search(searchTerm);
+    const searchResults = await wmSearch.search(searchTerm, searchId);
     return searchResults.filter((item) => item.type === walkme.content.TypeNames.SearchResult);
   } catch (err) {
     console.error('Could not get search results from the API.', err);
   }
 }
+
+export const getGuid = () => {
```
Comment (DavidHandler) [inline src/components/SearchResults/helpers.js:48]: Doesn't this function exist on a shared utils package/file?

File: src/components/TaskList/TaskItem.js [hunk 3]
```diff
@@ -4,11 +4,15 @@ import walkme from '@walkme/sdk';
...
 import { ReactComponent as TaskCheckbox } from './icons/taskCheckbox.svg';
 import { ReactComponent as TaskDone } from './icons/taskDone.svg';
+import { sendUIClickAnalytics } from '../SearchResults/helpers';
 
 import classes from './styles.module.scss';
 
-export default function TaskItem({ node }) {
+export default function TaskItem({ node, searchData }) {
   function onClick() {
+    const { searchId, searchTerm, totalResults, index } = searchData;
```
Comment (DavidHandler) [inline src/components/TaskList/TaskItem.js:13]: Don't we need a null check here?
Would we always have a search data when display a `TaskItem` (same for `HelpItem`) ?

File: src/components/SearchResults/SearchResultItem.js (hunk 1 above)
Comment (shlomi-d-walkme) [reply src/components/SearchResults/SearchResultItem.js:7]: We need to rethink and redesign the search app.
I think that there should be some SearchResultItems that the providers will be able to use

File: src/components/TaskList/TaskItem.js (hunk 3 above)
Comment (shlomi-d-walkme) [reply src/components/TaskList/TaskItem.js:13]: This was the bug 👍

File: src/components/SearchResults/helpers.js (hunk 2 above)
Comment (shlomi-d-walkme) [reply src/components/SearchResults/helpers.js:48]: It should. We will do it on the new search app

**Pull request merged**